File: frontend/src/types/cropAdvisory.ts
export interface CropAdvisoryRequest {
  soil_type?: string;
  irrigation_type?: string;
  location?: string;
  farm_size?: string;
  primary_crop?: string;
}

export interface RecommendationItem {
  crop: string;
  match_score: number;
  confidence: string;
  reasons: string[];
  water_requirement: string;
  season?: string | null;
  cultivation_note: string;
}

export interface CropAdvisoryInputSummary {
  farmer_name: string;
  location: string;
  farm_size: string;
  soil_type: string;
  irrigation_type: string;
  primary_crop: string;
}

export interface CropAdvisoryResponse {
  success: boolean;
  recommendations: RecommendationItem[];
  input_summary: CropAdvisoryInputSummary;
  missing_parameters: string[];
  notes?: string;
}

File: backend/app/schemas/crop_advisory.py
from pydantic import BaseModel
from typing import List, Optional

class CropAdvisoryRequest(BaseModel):
    soil_type: Optional[str] = None
    irrigation_type: Optional[str] = None
    location: Optional[str] = None
    farm_size: Optional[str] = None
    primary_crop: Optional[str] = None

class RecommendationItem(BaseModel):
    crop: str
    match_score: int
    confidence: str
    reasons: List[str]
    water_requirement: str
    season: Optional[str] = None
    cultivation_note: str

class InputSummary(BaseModel):
    farmer_name: str
    location: str
    farm_size: str
    soil_type: str
    irrigation_type: str
    primary_crop: str

class CropAdvisoryResponse(BaseModel):
    success: bool
    recommendations: List[RecommendationItem]
    input_summary: InputSummary
    missing_parameters: List[str]
    notes: Optional[str] = None
